report/test/renderer/dom-test.js

```diff
@@ -153,4 +152,73 @@ describe('DOM', () => {
           '<code>more code</code>, and yet event <code>more</code>.');
     });
   });
+
+  describe('safelySetHref', () => {
+    it('sets href for safe destinations', () => {
+      [
+        'https://safe.com/',
+        'http://safe.com/',
+      ].forEach(url => {
+        const a = dom.createElement('a');
+        dom.safelySetHref(a, url);
+        expect(a.href).toEqual(url);
+      });
+    });
+
+    it('sets href for safe in-page named anchors', () => {
+      const a = dom.createElement('a');
+      dom.safelySetHref(a, '#footer');
+      expect(a.href).toEqual('about:blank#footer');
+    });
+
+    it('doesnt set href if destination is unsafe', () => {
+      [
+        'javascript:evil()',
+        'data:text/html;base64,abcdef',
+        'data:application/json;base64,abcdef',
+        'ftp://evil.com/',
+        'blob:http://example.com/ca6df701-9c67-49fd-a787',
+        'intent://example.com',
+        'filesystem:http://localhost/img.png',
+        'ws://evilchat.com/',
+        'wss://evilchat.com/',
+        'about:about',
+        'chrome://chrome-urls/',
+      ].forEach(url => {
+        const a = dom.createElement('a');
+        dom.safelySetHref(a, url);
+
+        expect(a.href).toEqual('');
+        expect(!a.href).toBeTruthy();
+        expect(a.getAttribute('href')).toEqual(null);
+      });
+    });
+  });
+
+
+  describe('safelySetBlobHref', () => {
+    it('sets href for safe blob types', () => {
+      [
+        new window.Blob([JSON.stringify({test: 1234})], {type: 'application/json'}),
+        new window.Blob([`<html><h1>hello`], {type: 'text/html'}),
+      ].forEach(blob => {
+        const a = dom.createElement('a');
+        dom.safelySetBlobHref(a, blob);
+        expect(a.href).toEqual('null/blahblahblobid');
```

Comment:
use a jest.fn() for createObjectURL? `null` starting looks a bit like a bug in passing :)

Reply:
using jest.fn now